usaspending_api/data_load/derived_field_functions_fpds.py

```diff
@@ -0,0 +1,37 @@
+from datetime import datetime, timezone
+
+from usaspending_api.common.helpers.date_helper import fy
+from usaspending_api.data_load.cached_reference_data import subtier_agency_list
+from usaspending_api.broker.helpers.get_business_categories import get_business_categories
+
+
+def calculate_fiscal_year(broker_input):
+    return fy(broker_input["action_date"][:10])
+
+
+def calculate_awarding_agency(broker_input):
+    awarding_agency = subtier_agency_list().get(broker_input["awarding_sub_tier_agency_c"], None)
+    if awarding_agency is not None:
+        return awarding_agency["id"]
+    else:
+        return None
+
+
+def calculate_funding_agency(broker_input):
+    funding_agency = subtier_agency_list().get(broker_input["funding_sub_tier_agency_co"], None)
+    if funding_agency is not None:
+        return funding_agency["id"]
+    else:
+        return None
+
+
+def unique_transaction_id(broker_input):
+    return broker_input["detached_award_proc_unique"]
+
+
+def current_datetime(broker_input):
```

Comment:
I can't think of enough benefits from either approach to make it the clear choice. Personally the real-time/rolling is my preference

Reply:
Realtime is better, because that's when the value was actually updated. Broker chose the other option, which combined with the fact that submissions can take hours means that we can't use updated_at to compare against the last updated date to figure out when it was actually updated, meaning our FABS loader has to offset by an entire day, doubling its runtime.

Reply:
Realtime is better, because that's when the value was actually updated. Broker chose the other option, which combined with the fact that submissions can take hours means that we can't use updated_at to compare against the last updated date to figure out when it was actually updated, meaning our FABS loader has to offset by an entire day, doubling its runtime.

Reply:
@BrianZito FABS loader only looks back [15 minutes](https://github.com/fedspendingtransparency/usaspending-api/blob/dev/usaspending_api/broker/management/commands/fabs_nightly_loader.py#L20)

Reply:
We might want a conversation with Gollum about that then, since they confirmed to me that this would be a problem.